jetty-core/jetty-websocket/jetty-websocket-core-tests/src/test/java/org/eclipse/jetty/websocket/core/server/WebSocketServerTest.java

```diff
@@ -110,7 +110,8 @@ protected void demand()
             Frame frame = serverHandler.receivedFrames.poll(250, TimeUnit.MILLISECONDS);
             assertNull(frame);
 
-            serverHandler.getCoreSession().demand(2);
+            serverHandler.getCoreSession().demand();
+            serverHandler.getCoreSession().demand();
```

Comment:
This seems wrong, as the second call should throw.
If it's not causing flakyness, needs a comment explaining why.

Reply:
See comments from https://github.com/eclipse/jetty.project/pull/9945#discussion_r1241862275

I have opened https://github.com/eclipse/jetty.project/issues/9965 to change the implementation.

Reply:
In the next PR these should fail and I will need to change them.

Hopefully that will make beta3 as well, but if not at least we have the API changed with this.